src/main/scala/is/hail/expr/AST.scala

```diff
@@ -998,6 +998,14 @@ case class ApplyMethod(posn: Position, lhs: AST, method: String, args: Array[AST
               case (_: TArray, "forall") =>
                 val v = ir.genUID()
                 ir.ArrayFold(a, ir.True(), v, name, ir.ApplySpecial("&&", FastSeq(ir.Ref(v, TBoolean()), b)))
+              case (_: TSet, "map") => ir.ToSet(ir.ArrayMap(ir.ToArray(a), name, b))
+              case (_: TSet, "filter") => ir.ToSet(ir.ArrayFilter(ir.ToArray(a), name, b))
```

Comment:
will this sort again?

Reply:
there doesn't seem to be a way around that. non-fundamental types in the IR really feels wrong.

Reply:
I think we'll eventually have multiple levels of IR, and non-fundamental types would probably only exist in the highest level. The pass that compiles away non-fundamental types could see that all operations between `ToArray` and `ToSet` are order-preserving and omit the sort at the end.

In the meantime, if this becomes a problem we could add a parameter to `ToSet` that asserts the input is already sorted.